### PR TITLE
build: go module support

### PR DESCRIPTION
**Pull request by dajohi**
This is an attempt at a simple version of #341 

I have adjusted the import paths and upgraded the build scripts to use 1.13.8 and tried to keep the build script changes to a minimum.

I added a GitHub workflow for automated build and testing.  It currently runs `make all-bitcoin all-decred` and `make test`

https://github.com/dajohi/blockbook/actions/runs/46778188

### CODE REVIEW

File: build/docker/bin/Dockerfile [hunk 1]
```diff
@@ -33,17 +33,17 @@ RUN strip /opt/rocksdb/ldb /opt/rocksdb/sst_dump && \
...
 # download pre-loaded depencencies
 RUN \
-    cleanup() { rm -rf $GOPATH/src/blockbook; } && \
+    cleanup() { rm -rf $GOPATH/src/github.com/trezor ; } && \
     trap cleanup EXIT && \
-    cd $GOPATH/src && \
-    git clone https://github.com/trezor/blockbook.git && \
+    mkdir -p $GOPATH/src/github.com/trezor && \
+    cd $GOPATH/src/github.com/trezor && \
+    git clone -b gomod https://github.com/dajohi/blockbook.git && \
```
Comment (dajohi) [inline build/docker/bin/Dockerfile]: will need to be changed before merge

**Comment by martinboehm** [comment]
Hi. Thank you, it looks good at the first glance.
My only objection is the GitHub workflow: It is not related to the go module support.

**Comment by dajohi** [comment]
rebased removing the workflow

**Comment by martinboehm** [comment]
Thanks a lot for the fast update.
I just want to inform you that we will first make a new release of Blockbook - version 0.3.2 (very soon) and after that I will check and merge your PR. As this PR is a big infrastructure change, I don't want to push it to already well tested code. Instead, we will make it as a base of the next release.

**Comment by martinboehm** [comment]
Hi. Due to merge conflicts and also the necessity to test the PR changes, I merged your changes into a new branch called gomod. I will close this PR but your commits are already there, do not worry :)

File: api/types.go [hunk 2]
```diff
@@ -1,14 +1,15 @@
 package api
 
 import (
-	"blockbook/bchain"
```
Comment (IKE471) [inline api/types.go:4]: 4,5,6, red

File: Gopkg.lock [hunk 3]
```diff
@@ -1,298 +0,0 @@
...
-    "github.com/martinboehm/btcutil/txscript",
-    "github.com/martinboehm/golang-socketio",
-    "github.com/martinboehm/golang-socketio/transport",
-    "github.com/pebbe/zmq4",
-    "github.com/prometheus/client_golang/prometheus",
-    "github.com/prometheus/client_golang/prometheus/promhttp",
-    "github.com/schancel/cashaddr-converter/address",
-    "github.com/tecbot/gorocksdb",
-  ]
-  solver-name = "gps-cdcl"
```
Comment (IKE471) [inline Gopkg.lock:297]: all red

File: Gopkg.lock [hunk 4]
```diff
@@ -1,298 +0,0 @@
...
-  solver-version = 1
```
Comment (IKE471) [inline Gopkg.lock:298]: red

File: Gopkg.toml [hunk 5]
```diff
@@ -1,74 +0,0 @@
-
```
Comment (IKE471) [inline Gopkg.toml:1]: all red

File: api/typesv1.go [hunk 6]
```diff
@@ -1,8 +1,9 @@
 package api
 
 import (
-	"blockbook/bchain"
```
Comment (IKE471) [inline api/typesv1.go:4]: red

File: api/xpub.go [hunk 7]
```diff
@@ -1,8 +1,6 @@
 package api
 
 import (
-	"blockbook/bchain"
```
Comment (IKE471) [inline api/xpub.go:4]: 4,5 red

File: bchain/coins/bch/bcashparser.go [hunk 8]
```diff
@@ -1,15 +1,15 @@
 package bch
 
 import (
-	"blockbook/bchain"
```
Comment (IKE471) [inline bchain/coins/bch/bcashparser.go:4]: 4,5 red

**Pull request closed without merging**